src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java

```diff
@@ -16,15 +18,31 @@
 public class ZipStatusController {
 
     private final ZipFileStatusService service;
+    private static final int MIN_DCN_LENGTH = 6;
 
     // region constructor
+
     public ZipStatusController(ZipFileStatusService service) {
         this.service = service;
     }
+
     // endregion
 
     @GetMapping
-    public ZipFileStatus findByFileName(@RequestParam("name") String fileName) {
-        return service.getStatusFor(fileName);
+    public ResponseEntity getStatusByFilter(@RequestParam Map<String,String> filtersList) {
+
+        if (filtersList.keySet().contains("name") && filtersList.size() == 1 && !filtersList.get("name").isEmpty()) {
+            return ResponseEntity.ok(service.getStatusFor(filtersList.get("name")));
```

Comment:
The service method should be renamed to getStatusByName (or getStatusByFileName) - similar to getStatusByDcn